### ISSUE TITLE
develop initial model interface

### ISSUE THREAD

**Issue by hstojic** [hstojic/deep_bayesopt#2]
we are using neural nets as (approximate) probabilistic bayesian models, which means that the usual Keras methods might not be sufficient or incorrect, e.g. with MC dropout predict function is actually a single forward pass, which would be akin to a single sample

some methods that I think we should prescribe:
- predict_f and predict_y - this comes from gpflow, but it might be useful here as well, as with deep ensemble and MC dropout we can only do predict_y, but with deep evidential regression we can do both 
- sample - you expect to be able to sample from a probabilistic model 
- log likelihood - probably too much to ask...

any other?

of course, we can do this iteratively, as we add models and figure out shared methods/properties...

**Comment by hstojic** [comment]
@willcowley @SebastianPopescu any feedback here?

**Comment by hstojic** [comment]
PR is up now...
